### PR TITLE
Update `pgx` library to v5.

### PR DESCRIPTION
**Pull request by Tener**
Targeting release v15, so no backport to v14 or earlier.

### CODE REVIEW

**Review by ptgott — APPROVED** [review]
Approving for the `docs/pages` changes

**Review by rosstimothy — APPROVED** [review]

**Review by greedy52 — APPROVED** [review]

File: lib/srv/db/postgres/engine.go [hunk 1]
```diff
@@ -347,7 +353,8 @@ func (e *Engine) receiveFromClient(client *pgproto3.Backend, server *pgproto3.Fr
 			clientErrCh <- nil
 			return
 		}
-		err = server.Send(message)
+		server.Send(message)
+		err = server.Flush()
 		if err != nil {
```
[the remark this answers is not in the document]
Comment (Tener) [reply lib/srv/db/postgres/engine.go:358]: Thanks, but I wanted this change to be as easy to review as possible, so this means reducing the scope of changes. These changes are "nice to have", so I'm going to skip them in this PR.

**Review by smallinsky — APPROVED** [review]

**Comment by Tener** [comment]
I'm getting integration test failures, but only on CI. All of them are complaining about missing data, like this:

```code
    db_integration_test.go:180: 
        	Error Trace:	/__w/teleport/teleport/integration/db/db_integration_test.go:180
        	Error:      	Not equal: 
        	            	expected: []*pgconn.Result{(*pgconn.Result)(0x12b6f6e0)}
        	            	actual  : []*pgconn.Result{(*pgconn.Result)(0xc003a7b720)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,20 +2,4 @@
        	            	  (*pgconn.Result)({
        	            	-  FieldDescriptions: ([]pgconn.FieldDescription) (len=1) {
        	            	-   (pgconn.FieldDescription) {
        	            	-    Name: (string) (len=10) "test-field",
        	            	-    TableOID: (uint32) 0,
        	            	-    TableAttributeNumber: (uint16) 0,
        	            	-    DataTypeOID: (uint32) 0,
        	            	-    DataTypeSize: (int16) 0,
        	            	-    TypeModifier: (int32) 0,
        	            	-    Format: (int16) 0
        	            	-   }
        	            	-  },
        	            	-  Rows: ([][][]uint8) (len=1) {
        	            	-   ([][]uint8) (len=1) {
        	            	-    ([]uint8) (len=10) {
        	            	-     00000000  74 65 73 74 2d 76 61 6c  75 65                    |test-value|
        	            	-    }
        	            	-   }
        	            	-  },
        	            	+  FieldDescriptions: ([]pgconn.FieldDescription) <nil>,
        	            	+  Rows: ([][][]uint8) <nil>,
        	            	   CommandTag: (pgconn.CommandTag) {
        	Test:       	TestDatabaseAccess/IPPinning/root_cluster,_no_pinned_ip
```

This is very reliable on CI, failing on each run (although in a different test), but very hard to replicate locally. Thoughts?

**Comment by greedy52** [comment]
> This is very reliable on CI, failing on each run (although in a different test), but very hard to replicate locally. Thoughts?

Can't repro locally either. But based on the failure log, at least the test server is sending the response. Maybe enable the log to check if db engine is sending back to client?

**Comment by Tener** [comment]
> > This is very reliable on CI, failing on each run (although in a different test), but very hard to replicate locally. Thoughts?
> 
> Can't repro locally either. But based on the failure log, at least the test server is sending the response. Maybe enable the log to check if db engine is sending back to client?

I've added some debugging code and run it lots of times, and the issue is strange. It does occur for me locally with unmodified code, but rarely. Sometimes the messages before the "end of query" just disappear - the library is unaware of them, even though there are logs showing they were sent - but it is inconsistent: the number of messages disappearing changes.

I _think_ this is most likely a latent bug in `pgx` which somehow surfaces due to the way we are using the library, but I'd like to confirm it in some way.

**Comment by ptgott** [comment]
@Tener Circling back to this one. Are we planning to merge it closer to the v15 release?

**Comment by Tener** [comment]
> @Tener Circling back to this one. Are we planning to merge it closer to the v15 release?

@ptgott sorry, this slipped my notice. The upgrade causes some issues which needs investigating before we can safely proceed. I'm closing this for now.

**Pull request closed without merging**